server/routes/finance-id-bank-account/financeIdBankAccountController.test.ts

```diff
@@ -0,0 +1,216 @@
+  })
+})
+describe('getUpdateBankAccountStatusView', () => {
+  it('Happy path update bank account', async () => {
+    await request(app)
+      .get('/finance-and-id/add-a-bank-account/?prisonerNumber=123&applicationId=123')
+      .expect(200)
+      .expect(res => expect(res.text).toMatchSnapshot())
+  })
+  it('error, no prisoner number', async () => {
+    await request(app)
+      .get('/finance-and-id/add-a-bank-account/?prisonerNumber=&applicationId=123')
+      .expect(500)
+      .expect(res => expect(sanitiseStackTrace(res.text)).toMatchSnapshot())
+  })
+  it('no application ID', async () => {
+    await request(app)
+      .get('/finance-and-id/add-a-bank-account/?prisonerNumber=123&applicationId=')
+      .expect(200)
+      .expect(res => expect(sanitiseStackTrace(res.text)).toMatchSnapshot())
+  })
+})
+describe('getConfirmAddABankAccountView', () => {
+  it('Happy path confirm add bank account', async () => {
+    await request(app)
+      .get(
+        '/finance-and-id/confirm-add-a-bank-account/?bankName=Lloyds&applicationSubmittedDay=01&applicationSubmittedMonth=01&applicationSubmittedYear=01&prisonerNumber=123&applicationId=&confirmationType=addAccount&applicationType=',
+      )
+      .expect(200)
+      .expect(res => expect(res.text).toMatchSnapshot())
+  })
+  it('error, no prisoner number', async () => {
+    await request(app)
+      .get(
+        '/finance-and-id/confirm-add-a-bank-account/?bankName=Lloyds&applicationSubmittedDay=01&applicationSubmittedMonth=01&applicationSubmittedYear=01&applicationId=&confirmationType=addAccount&applicationType=',
+      )
+      .expect(500)
+      .expect(res => expect(sanitiseStackTrace(res.text)).toMatchSnapshot())
+  })
+  it('only prisoner number, no day/month/year/bank name', async () => {
+    await request(app)
+      .get(
+        '/finance-and-id/confirm-add-a-bank-account/?prisonerNumber=123&applicationId=&confirmationType=addAccount&applicationType=',
+      )
+      .expect(200)
+      .expect(res => expect(sanitiseStackTrace(res.text)).toMatchSnapshot())
+  })
+})
+describe('postBankAccountSubmitView', () => {
+  it('happy path - posting bank account', async () => {
+    const postBankApplicationSpy = jest.spyOn(rpService, 'postBankApplication').mockResolvedValue({})
+    await request(app)
+      .post('/finance-and-id/bank-account-submit')
+      .send({
+        prisonerNumber: '123',
+        applicationDate: '01/01/2024',
+        bankName: 'BARCLAYS',
+      })
+      .expect(302)
+      .expect(res => expect(res.text).toEqual('Found. Redirecting to /finance-and-id/?prisonerNumber=123#finance'))
+    expect(postBankApplicationSpy).toHaveBeenCalledWith('123', {
+      applicationSubmittedDate: '2024-01-01T00:00:00',
+      bankName: 'BARCLAYS',
+    })
+  })
+  it('error - missing prisoner number', async () => {
+    await request(app)
+      .post('/finance-and-id/bank-account-submit')
+      .send({
+        prisonerNumber: '',
```

Comment:
Might be worth checking the scenarios when they are missing then rename these as blank.